iroh-gateway/src/headers.rs

```diff
@@ -10,16 +10,34 @@ pub fn add_user_headers(headers: &mut HeaderMap, user_headers: HeaderMap) {
 }
 
 #[tracing::instrument()]
-pub fn add_content_type_headers(headers: &mut HeaderMap, name: &str) {
+pub fn add_content_type_headers(headers: &mut HeaderMap, name: &str, body_sample: &[u8]) {
     let guess = mime_guess::from_path(name);
-    if let Some(content_type) = guess.first() {
+    let mut content_type = String::new();
+    if let Some(ct) = guess.first() {
+        content_type = ct.to_string();
+    } else {
         // todo(arqu): deeper content type checking
-        // todo(arqu): if mime type starts with text/html; strip encoding to let browser detect
-        headers.insert(
-            CONTENT_TYPE,
-            HeaderValue::from_str(content_type.as_ref()).unwrap(),
+        let classifier = mime_classifier::MimeClassifier::new();
```

Comment:
could you store & reuse this classifier?

Reply:
It's neither large nor stateful really, we save nothing and I have to bother with a fork to add clone to it. Also having a global one would mean we have one of these cloned/instantiated per req. This way it's only when the req needs it.